### PR TITLE
ensuring sssd.conf consistency during first Puppet run

### PR DESCRIPTION
**Pull request by curly8**

### CODE REVIEW

**Comment by edestecd** [comment]
Interesting, I didn't think of authconfig modifying the config file.  That makes sense though.

Can you add a line to the spec test that makes sure this order happens in the RedHat blocks...

**Comment by curly8** [comment]
All check finally passed!

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -11,6 +11,9 @@
       it { is_expected.to contain_package('authconfig').with_ensure('latest') }
       it { is_expected.not_to contain_package('oddjob-mkhomedir') }
       it { is_expected.not_to contain_service('oddjobd') }
+      it 'should pass' do
+        contain_exec('authconfig-mkhomedir').that_comes_before('File[sssd.conf]')
```
Comment (edestecd) [inline spec/classes/init_spec.rb]: I think you need a ```is_expected.to``` here:
```ruby
it 'should pass' do
  is_expected.to contain_exec('authconfig-mkhomedir').that_comes_before('File[sssd.conf]')
end
```

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -31,6 +34,9 @@
       it { is_expected.to contain_package('authconfig').with_ensure('present') }
       it { is_expected.to contain_package('oddjob-mkhomedir') }
       it { is_expected.to contain_service('oddjobd') }
+      it 'should pass' do
+        contain_exec('authconfig-mkhomedir').that_comes_before('File[sssd.conf]')
```
Comment (edestecd) [inline spec/classes/init_spec.rb]: I think you need a ```is_expected.to``` here:
```ruby
it 'should pass' do
  is_expected.to contain_exec('authconfig-mkhomedir').that_comes_before('File[sssd.conf]')
end
```

**Comment by edestecd** [comment]
If you want to bump the line length requirement just a bit, I wouldn't be opposed:
Its set here:
https://github.com/sgnl05/sgnl05-sssd/blob/master/.rubocop.yml#L26

I often bump it to 100

**Comment by edestecd** [comment]
@curly8 can you add the same line to the RedHat 7 block also?

**Comment by curly8** [comment]
Done. Had to bump the Line Length to 105. Please review.

**Comment by edestecd** [comment]
Looks good

**Pull request merged**